backend/models/distance.py

```diff
@@ -217,9 +225,76 @@ def get_matrix(self, source_points, destination_points):
         )
 
         response.raise_for_status()
+
         content = response.json()
         return content['durations']
 
+    def measure_one_to_many_points(self, origin, point_list):
+        """
+        Measure duration from one to mamy points using the matrix API.
+
+        Result is a list of duration in  minutes.
+        """
+        measurements = []
+
+        for batch in iterators.iterate_in_slices(
+            iterable=iter(point_list),
+                batch_size=24):
+            measurements.extend(
+                self.get_matrix(
+                    source_points=[origin],
+                    destination_points=batch)[0]
+            )
+        return measurements
+
+    def closest_with_early_exit(self, origin, point_list, exit_distance):
+        """
+        Find closest point in a list of points, exiting early if exit_distance is reached.
+
+        The exit_distance uses haversine distance (in meters).
+        """
+        haversine_distances = [
+            self._haversine_measurer.measure_between_two_points(origin, point)
+            for point in point_list
+        ]
+        min_haversine_idx, min_haversine_distance = min(
+            enumerate(haversine_distances),
+            key=operator.itemgetter(1)
+        )
+
+        # TODO - Avoid early 'return' so that we can implement one call to
```

Comment:
I do not understand this TODO. We are already within the closest with early exit function, and this does use closest.